### PR TITLE
Adds recipe for NGram

### PR DESCRIPTION
**Pull request by kylebgorman**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by kylebgorman** [comment]
Hi @conda-forge/help-c-cpp, when you have time, would appreciate any advice on the failure in the staging stuff. One log says that a `conda_build.exceptions.OverLinkingError` was thrown: not sure what to do about this.

File: recipes/ngram/build.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+CPPFLAGS="-I${CONDA_PREFIX}/include"
+CXXFLAGS="-L${CONDA_PREFIX}/lib"
```
Comment (isuruf) [inline recipes/ngram/build.sh]: Remove the 2 lines above

Comment (kylebgorman) [reply recipes/ngram/build.sh]: Done, thanks.

File: recipes/ngram/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    - {{ compiler('cxx') }}
+    - make
+  host:
+    - openfst ==1.7.5
+  run:
+    - openfst ==1.7.5
+
+test:
+  requires:
+    - openfst ==1.7.5
```
Comment (isuruf) [inline recipes/ngram/meta.yaml]: Can you remove the 2 lines above?

Comment (kylebgorman) [reply recipes/ngram/meta.yaml]: Done; tests look good locally.

**Pull request merged**